doc/development_guide/contributor_guide/governance.rst

```diff
@@ -0,0 +1,111 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Contribute for more than 3 releases consistently.
+- Do not be too opinionated, follow the code of conduct without requiring emotional
+  works from the maintainers. It does not mean that disagreements are impossible,
+  only that arguments should stay technical and reasonable so the conversation
+  is civil and productive.
+- Have a maintainer suggest that you become triager, without you asking
+- Get unanimous approval or neutral agreement from current maintainers.
+
+How to become a maintainer ?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Contribute two big code merge requests over multiple releases (for example
+  one checker in 2.13 and the following bug after release and one complicated
+  bug fixes in 2.14). Otherwise contributing for more than 3 releases consistently
+  with great technical and interpersonal skills.
+- Take ownership of a part of the code that is not maintained well at the moment
+  or that you contributed personally (if we feel we can't merge something without
+  your review, you're going to be able to merge those yourself soon).
+- Triage for multiple months (close duplicate, clean up issues, answer questions...)
+- Have an admin suggest that you become maintainer, without you asking
+- Get unanimous approval or neutral agreement from current maintainers.
+
+
+How to become an admin ?
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Contribute for several hundreds of commits over a long period of time
+  with excellent interpersonal skills and code quality.
+- Maintain pylint for multiple years (code review, triaging and maintenance tasks).
+- At this point probably have another admin leave the project or
+  become inactive for years.
+- Have an admin suggest that you become an admin, without you asking.
+- Get unanimous approval or neutral agreement from current admins.
+
+
+How are decisions made ?
+------------------------
+
+Everyone is expected to follow the code of conduct. pylint is a do-ocracy / democracy.
+You're not allowed to behave poorly because you contributed a lot. But if
+you're not going to do the future maintenance work, your valid opinions might not be
+taken into account by those that will be affected by it.
+
+What are the fundamental of pylint ?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+General:
+
+- We favor correctness over performance, because pylint is not used primarily
+  for its performance. Performance is still important and need to be taken into
+  account from the get go.
+
+- We then favor false negatives over false positives if correctness is
+  impossible to achieve.
+
+- We try to keep the configuration sane, but if there's a hard decision to take we
+  add an option so that pylint is multiple sizes fit all (after configuration)
+
+Where to add a new checker or message:
+
+- Error messages (thing that will result in an error if run) should be builtin
+  checks, activated by default
+
+- Messages that are opinionated, even slightly, should be opt-in (added as an extension)
```

Comment:
You might consider linking to the docs for adding extensions, but it's not essential.

Reply:
[d357be6](https://github.com/PyCQA/pylint/pull/8329/commits/d357be6f9c9b52d5eb40b049bf72e11464102a9d)